build-system/tasks/storybook-react/index.js

```diff
@@ -0,0 +1,104 @@
+('use strict');
+
+const argv = require('minimist')(process.argv.slice(2));
+const path = require('path');
+const {build: buildBinary} = require('../build');
+const {createCtrlcHandler} = require('../../common/ctrlcHandler');
+const {cyan} = require('kleur/colors');
+const {defaultTask: runAmpDevBuildServer} = require('../default-task');
+const {exec, execScriptAsync} = require('../../common/exec');
+const {isCiBuild} = require('../../common/ci');
+const {log} = require('../../common/logging');
+
+const ENV_PORTS = {
+  amp: 9001,
+  preact: 9002,
+  react: 9003,
+};
+
+const repoDir = path.join(__dirname, '../../..');
+
+/** @typedef {'amp'|'preact'|'react'} StorybookEnv */
+
+/**
+ * @param {StorybookEnv} env
+ * @return {string}
+ */
+const envConfigDir = (env) => path.join(__dirname, `${env}-env`);
+
+/**
+ * @param {StorybookEnv} env
+ */
+function launchEnv(env) {
+  log(`Launching storybook for the ${cyan(env)} environment...`);
+  const {'storybook_port': port = ENV_PORTS[env]} = argv;
+  execScriptAsync(
+    [
+      './node_modules/.bin/start-storybook',
+      `--config-dir ${envConfigDir(env)}`,
+      `--static-dir ${repoDir}/`,
+      `--port ${port}`,
+      '--quiet',
+      isCiBuild() ? '--ci' : '',
+    ].join(' '),
+    {cwd: __dirname, stdio: 'inherit'}
+  ).on('error', () => {
+    throw new Error('Launch failed');
+  });
```

Comment:
The Error class seems to only take one argument, which is a string.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#:~:text=When%20catching%20and%20re%2Dthrowing%20an%20error%20with%20a%20more%2Dspecific%20or%20useful%20error%20message%2C%20this%20property%20should%20be%20used%20to%20pass%20the%20original%20error.

It has a few signatures. Looks like I was also wrong; should be
```
```suggestion
  ).on('error', (err) => {
    throw new Error('Launch failed', {cause: err});
  });
```

Reply:
I'm getting is typescript error
```
Expected 0-1 arguments, but got 2.ts(2554)
```
but i'm just going to ignore it since our linter doesn't seem to mind.

Reply:
Nice find! It appears MS Edge doesn't support this param:
![image](https://user-images.githubusercontent.com/6694512/150416730-93e427b0-52b1-45cd-b4e0-ff6310929123.png)

Maybe the best we can do here then is
`throw new Error(`Launch failed: ${err}`);`